modular_zubbers/code/modules/rimpoint_newfeatures/code/umbrella.dm

```diff
@@ -0,0 +1,146 @@
+/obj/item/umbrella
+	name = "umbrella"
+	icon = 'modular_zubbers/icons/obj/equipment/umbrella.dmi'
+	icon_state = "umbrella"
+	lefthand_file = 'modular_zubbers/icons/obj/equipment/umbrella_inhand_l.dmi'
+	righthand_file = 'modular_zubbers/icons/obj/equipment/umbrella_inhand_r.dmi'
+	inhand_icon_state = "umbrella"
+	w_class = WEIGHT_CLASS_BULKY
+	custom_materials = list(/datum/material/iron = SMALL_MATERIAL_AMOUNT * 0.75, /datum/material/plastic = SMALL_MATERIAL_AMOUNT * 0.25)
+	custom_premium_price = PAYCHECK_COMMAND * 2
+
+	// Sound effect from the rain hitting it.
+	var/datum/looping_sound/umbrella/rain_on_plastic/looping_sound
+
+	// Seems easier to do this.
+	var/mob/user_holding
+
+	// For some overhang.
+	inhand_x_dimension = 64
+	inhand_y_dimension = 64
+
+	// The umbrella provides some protection against weather effects. Perhaps you may want to change this. I don't judge.
+	var/immunity_type = TRAIT_RAINSTORM_IMMUNE
+
+
+/obj/item/umbrella/equipped(mob/user, slot, initial = FALSE)
+	. = ..()
+	if(!looping_sound)
+		looping_sound = new(src, FALSE)
+	if(!user_holding)
+		RegisterSignal(user, COMSIG_MOVABLE_MOVED, PROC_REF(on_move))
+	user_holding = user
+	proccess_rainsounds()
+
+/obj/item/umbrella/dropped(mob/user, silent)
+	. = ..()
+	revoke_trait(user, immunity_type)
+	QDEL_NULL(looping_sound)
+	UnregisterSignal(user, COMSIG_MOVABLE_MOVED)
+	user_holding = null
+
+/obj/item/umbrella/proc/on_move(atom/old_loc, dir, forced, list/old_locs)
+	SIGNAL_HANDLER
+
+	proccess_rainsounds(old_loc)
+
+/obj/item/umbrella/proc/proccess_rainsounds()
+	if(!SSweather)
+		return
```

Comment:
if you're gonna want to return a boolean ensure you always return a boolean, or just don't do booleans if you don't make use of it or have a reason for i t